src/wp-includes/class-wp-post-type.php

```diff
@@ -922,7 +922,7 @@ public function get_autosave_rest_controller() {
 			return null;
 		}
 
-		if ( 'attachment' === $this->name ) {
+		if ( in_array( $this->name, array( 'attachment', 'wp_global_styles' ), true ) ) {
```

Comment:
I like this approach. Simple!

`* @since 6.5.0 Prevents autosave class instantiation for `wp_global_styles` post types.`